src/modules/sync-cli.js

```diff
@@ -781,9 +762,84 @@ var self = {
     });
   },
 
-  existingPendingPreHashMap: function(dataset){
-    var pendingPreHashes = dataset.changeHistory || {};
-    return pendingPreHashes;
+  applyPendingChangesToRecords: function(dataset, records){
+    var pendings = dataset.pending;
+    for(var pendingUid in pendings){
+      if(pendings.hasOwnProperty(pendingUid)){
+        var pendingObj = pendings[pendingUid];
+        var uid = pendingObj.uid;
+        //if the records contain any thing about the data records that are currently in pendings,
+        //it means there are local changes that haven't been applied to the cloud yet,
+        //so update the pre value of each pending record to relect the latest status from cloud
+        //and remove them from the response
+        var keys = ['create', 'update', 'delete'];
+        for(var i = 0; i<keys.length;i++){
```

Comment:
@corinnekrych  you are talking about this for loop and if else statements here, right?